Mathlib/Combinatorics/SimpleGraph/Maps.lean

```diff
@@ -671,4 +672,17 @@ noncomputable def overFinIso (hc : Fintype.card V = n) : G ≃g G.overFin hc :=
 
 end Finite
 
+section degree
+
+variable (v) [Fintype (G.neighborSet v)] (f : G ≃g G')
+
+instance : Fintype (G'.neighborSet (f v)) :=
+  Fintype.ofEquiv ↑(G.neighborSet v) (f.mapNeighborSet v)
```

Comment:
This creates a diamond with the existing instance that gives `Fintype (G.neighborSet _)` from `Fintype G` and some `Decidable*`.

Reply:
What is a diamond and why is it problematic?